### PR TITLE
Use system time at start

### PR DESCRIPTION
**Pull request by fluffyfreak**
When starting a new Game offset the start time from the year 2000, specifically Friday, 31 December 1999 23:59:59 so that the date and time are exactly 1200 years from whenever they start the game.

This is only done when starting a new game, saved games are not affected.

When testing it today my starting date and time were: 3219, Nov 16, 14:47

### CODE REVIEW

**Comment by fluffyfreak** [comment]
Good point, guess that would need updating, wonder if there's any other hardcoded date entries it might mess up

File: src/Game.h [hunk 1]
```diff
@@ -170,7 +170,7 @@ class Game {
 
 	std::unique_ptr<Player> m_player;
 
-	enum State {
+	enum class State {
 		STATE_NORMAL,
 		STATE_HYPERSPACE,
```
Comment (impaktor) [inline src/Game.h]: I don't know what the naming convention is, but perhaps the variable don't need "STATE" prefix when it's given by the class, e.g. `state::STATE_NORMAL`?

File: src/LuaGame.cpp [hunk 2]
```diff
@@ -63,8 +63,13 @@ static int l_game_start_game(lua_State *l)
 	}
 
 	SystemPath *path = LuaObject<SystemPath>::CheckFromLua(1);
-	const double start_time = luaL_optnumber(l, 2, 0.0);
```
Comment (impaktor) [inline src/LuaGame.cpp]: A short API documentation compatible paragraph might be nice, or is this only for internal use? Any chance someone else might want to use this?

**Comment by impaktor** [comment]
I should say, I think checking system time might be fun/useful for easter eggs, like putting a santa hat on BBS NPCs if it's 24-26 december, and things like that.

File: src/LuaGame.cpp (hunk 2 above)
Comment (fluffyfreak) [reply src/LuaGame.cpp]: It's an old function that is documented just above. It's always been in there but I don't think anyone uses it.

File: src/Game.h (hunk 1 above)
Comment (fluffyfreak) [reply src/Game.h]: Good point I'll change that!

**Comment by fluffyfreak** [comment]
As for the forum, just change the question to ask what century? Then we'll be good for the next 81 years ;)

**Comment by fluffyfreak** [comment]
Or just change it every year, so this right now it'll be 3219, next year it will become 3220!

**Comment by zonkmachine** [comment]
> Or just change it every year, so this right now it'll be 3219, next year it will become 3220!

So this is why I had problems registering to the wiki and forum. :)
Maybe instead ask what ship you start with on Mars?

**Comment by impaktor** [comment]
@zonkmachine indeed, both wiki and forum need annual upadte of the security question. You've been activated on the forum now, btw.

**Pull request merged**